fdbcli/tests/metacluster_fdbcli_tests.py

```diff
@@ -42,157 +44,489 @@ def run_fdbcli_command(cluster_file, *args):
+        command = command + " assigned_cluster={}".format(assigned_cluster)
+
+    _, output, err = run_fdbcli_command(management_cluster_file, command)
+    return output, err
+
+
+def list_tenants(
+    management_cluster_file,
+    data_cluster_files,
+    tenant_name_begin=None,
+    tenant_name_end=None,
+):
+    command = "tenant list"
+    if tenant_name_begin:
+        command = command + " {}".format(tenant_name_begin)
+    if tenant_name_end:
+        command = command + " {}".format(tenant_name_end)
+    _, output, err = run_fdbcli_command(management_cluster_file, command)
+    return output, err
+
+
+def get_tenant_names(
+    management_cluster_file,
+    data_cluster_files,
+    tenant_name_begin=None,
+    tenant_name_end=None,
+):
+    command = "tenant list"
+    if tenant_name_begin:
+        command = command + " {}".format(tenant_name_begin)
+    if tenant_name_end:
+        command = command + " {}".format(tenant_name_end)
+    rc, output, err = run_fdbcli_command(management_cluster_file, command)
+    if rc != 0:
+        return []
+    print(output)
+    res = []
+    lines = output.split("\n")
+    for ln in lines:
+        m = re.match(r"^\d+\.\s+(?P<name>\w+)$", ln.strip())
+        if m:
+            res.append(m.group("name"))
+    return res
+
+
+def rename_tenant(management_cluster_file, data_cluster_files, old_name, new_name):
+    command = "tenant rename {old_name} {new_name}".format(
+        old_name=old_name, new_name=new_name
+    )
+    _, output, err = run_fdbcli_command(management_cluster_file, command)
+    return output, err
+
+
+def get_tenant(management_cluster_file, data_cluster_files, name):
+    command = "tenant get {}".format(name)
+    _, output, err = run_fdbcli_command(management_cluster_file, command)
+    return output, err
 
+
+def delete_tenant(management_cluster_file, data_cluster_files, name):
+    command = "tenant delete {}".format(name)
+    _, output, err = run_fdbcli_command(management_cluster_file, command)
+    return output, err
+
+
+def delete_tenant_by_id(management_cluster_file, data_cluster_files, id):
+    command = "tenant deleteId {}".format(id)
     _, output, err = run_fdbcli_command(management_cluster_file, command)
     return output, err
 
 
 def clear_database_and_tenants(management_cluster_file, data_cluster_files):
-    subcmd1 = 'writemode on'
-    subcmd2 = 'option on SPECIAL_KEY_SPACE_ENABLE_WRITES'
-    subcmd3 = 'clearrange ' '"" \\xff'
-    subcmd4 = 'clearrange \\xff\\xff/management/tenant/map/ \\xff\\xff/management/tenant/map0'
+    all_tenants = get_tenant_names(management_cluster_file, data_cluster_files)
+    for cf in data_cluster_files:
+        subcmd1 = "writemode on"
+        subcmd2 = "clearrange " '"" \\xff'
+        run_fdbcli_command(cf, subcmd1, subcmd2)
+    for tenant in all_tenants:
+        delete_tenant(management_cluster_file, data_cluster_files, tenant)
+    subcmd1 = "writemode on"
```

Comment:
We probably don't want to do these steps from the above line down in a metacluster (unlike in the non-metacluster tests) because this is better accomplished by unregistering the clusters and decommissioning the metacluster.

Reply:
SGTM. We already perform the cleanup in `cleanup_after_test`. I'll remove these lines from `clear_database_and_tenants` and rename the function to `clear_all_tenants`.